eesp-ikev2.org

```diff
@@ -82,154 +83,76 @@ described in BCP 14 [[RFC2119]] [[RFC8174]] when, and only when, they
-sequence numbers are disabled, there won't be any SN in the
-EESP packet, the receiver SHOULD NOT dynamically modify ports or
-addresses without using IKEv2 Mobility [[RFC4555]].
-
-Because the Replay Protection service is disabled, an attacker can re
-play packets with a different source address. Such an attacker could
-disrupt the connection by replaying a single packet with a different
-source address or port number.
-
-** Explicit Initialization Vector
-
-If the algorithm used to encrypt the payload requires cryptographic
-synchronization data, e.g., an Initialization Vector (IV), then this
-may be carried explicitly in every EESP packet.
-
-** Implicit Initialization Vectors
-
-With the Implicit Initialization Vector (IIV) encryption algorithm,
-as specified in [[RFC8750]], the IV MUST be omitted in the EESP
-packet. To enable this functionality, IIV transforms defined in
-[[IKEv2-Enc]] MUST be used during negotiation. Furthermore,
-the [[IKEv2-SN]] extension MUST be negotiated to support the use of
-64-bit Sequential Numbers in EESP packets. If the the proposal
-does not include 64-bit Sequential Numbers return error
-NO_PROPOSAL_CHOSEN.
+# [VS] See note above. This should either be a normative reference
+# [VS] And we use a different term - Sub SA. "Sub-Child SA" is not
+# [VS] used in the text. I'd rather to delete this
+# This document uses the following terms defined in
+# [[I-D.mrossberg-ipsecme-multiple-sequence-counters]]: Sub-Child SA.
+
+# [VS] This is the name of transform, I don't think we should 
+# [VS] reference ikev2-rename-esn here
+# [VS] ikev2-rename-esn is only relevant *now* until it becomes an 
+# [VS] RFC and IANA updates IKEv2 registries
+# [VS] For this reason it is referenced in G-IKEv2 draft, which uses 
+# [VS] not-yet-assigned name for this transform
+# [VS] But by the time this draft would be published, the 
+# [VS] ikev2-rename-esn will most probably become an RFC and IANA 
+# [VS] completes the renaming,
+# [VS] thus we can just reference the IANA registry
+# This document uses the following terms defined in
+# [[I-D.ietf-ipsecme-ikev2-rename-esn]] : Replay Protection.
+
+# [VS] I'd rather to put off all the group SA stuff 
+# This document uses the following terms defined in
+# [[I-D.ietf-ipsecme-g-ikev2]]: Sender-ID, Data-Security SA,
+# GWP_SENDER_ID_BITS, GCKS policy.
+
+* EESP Overview
 
 ** EESP Version
+
 Each EESP packets carry EESP Base Header version, which is specified
-[[I-D.klassert-ipsecme-eesp]]. This SHOULD BE negotiated using
-IKEv2. Each Base Header version, to be able to negotiate via IKEv2,
-SHOULD have a corresponding ~IKEv2 Security Protocol Identifiers~
-The initial version sepecified EESPv0(TBD1)
+in Section [XXX] of [[I-D.klassert-ipsecme-eesp]]. This SHOULD be 
+negotiated using IKEv2. Each Base Header version, to be able to 
+negotiate via IKEv2, SHOULD have a corresponding ~IKEv2 Security 
+Protocol Identifiers~ The initial version sepecified EESPv0 (TBD1)
```

Comment:
```suggestion
Protocol Identifiers~ The initial version specified is EESPv0 (TBD1)
```

I wonder why this would be SHOULD. Is that indented for authors of future versions/extensions? Because for an implementer of this protocol, how could it be SHOULD if we define the only way to negotiate it is with proposals  that carry the new IPsec protocol identifier we define? Maybe that part of the text should be placed in the EESP draft?

Reply:
This text was copy-pasted from early version, it needs to be edited. I agree that SHOULD is wrong here. The EESP version is negotiated as part of the negotiated process, it cannot be skipped, thus no "MUST", "SHOULD", "MAY" cannot be used here.